src/transformers/configuration_utils.py

```diff
@@ -422,6 +422,21 @@ def num_labels(self, num_labels: int):
             self.id2label = {i: f"LABEL_{i}" for i in range(num_labels)}
             self.label2id = dict(zip(self.id2label.values(), self.id2label.keys()))
 
+    @property
+    def attn_implementation(self):
+        if not hasattr(self, "_attn_implementation"):
+            return "eager"
+        else:
+            return self._attn_implementation
+
+    @attn_implementation.setter
+    def attn_implementation(self, value):
+        if hasattr(self, "attn_implementation_set") and self.attn_implementation_set:
+            raise NotImplementedError("Modifying the attention implementation through this attribute is currently not implemented.")
+        self.attn_implementation_set = True
+
+        self._attn_implementation = value
```

Comment:
Please comment on this. Is this fine to be in `PretrainedConfig`? Should this rather be in each of the `configuration_*.py` with `Copied from` statements? Should this rather be in `PreTrainedModel`?